demo/sinetiq-core-application-temperature-consumer/src/main/java/se/sinetiq/core/demo/application/consumer/TemperatureSensorConsumer.java

```diff
@@ -15,61 +15,77 @@
+        ConsulAPI consulAPI = new ConsulAPI();
+        try {
+            consulAPI.configure(host, port);
+        } catch (Throwable t) {
+            System.out.println("Service registry configuration failure!");
+            t.printStackTrace();
+            System.exit(1);
+        }
 
+        ServiceData sd = lookupTempService(consulAPI);
 
-    //
-    // Part 1: Discover (lookup) services (api instances) that fulfill the api specification 'temperature-sensor'
-    //
-    ServiceType serviceType = new ServiceType("temperature-sensor-rest-json");
-    List<ServiceName> apiInstances = consulAPI.getServiceInstances(serviceType);
-    System.out.println("Found " + apiInstances.size() + " instaces.");
-    for(ServiceName sn : apiInstances) {
-      // Found service
-      System.out.println("+ " + sn.getName());
-    }
+        ApiClient defaultClient = Configuration.getDefaultApiClient();
+        String pathStr = sd.getProperties().get("path") != null ? sd.getProperties().get("path") : "";
+        defaultClient.setBasePath("http://" + sd.getHost() + ":" + sd.getPort() /*+ pathStr*/);
 
-    if(apiInstances.size() <= 0) {
-      System.out.println("No services found, abort");
-      return;
-    }
+        DefaultApi apiInstance = new DefaultApi(defaultClient);
+        try {
+            TemperatureData result = apiInstance.temperatureGet();
+            System.out.println("Response from " + sd.getName() + ":");
+            System.out.println(result);
 
-    //
-    // Part 2: Use discovery endpoint details for establish connection
-    //
-    ServiceName sn = apiInstances.get(0); // Just pick the first service (api instance) found
-    ServiceData sd = consulAPI.getServiceData(sn);
-    if(sd == null) {
-      System.out.println("No services data found, abort");
-      return;
+        } catch (ApiException e) {
+            System.err.println("Exception when calling DefaultApi#temperatureGet");
+            System.err.println("Status code: " + e.getCode());
+            System.err.println("Reason: " + e.getResponseBody());
+            System.err.println("Response headers: " + e.getResponseHeaders());
+            /*e.printStackTrace();*/
+        }
     }
-    ApiClient defaultClient = Configuration.getDefaultApiClient();
-    String pathStr = sd.getProperties().get("path")!=null?sd.getProperties().get("path"):"";
-    defaultClient.setBasePath("http://" + sd.getHost() + ":" + sd.getPort() /*+ pathStr*/);
 
-    DefaultApi apiInstance = new DefaultApi(defaultClient);
-    try {
-      TemperatureData result = apiInstance.temperatureGet();
-      System.out.println(result);
+    static ServiceData lookupTempService(ConsulAPI consulAPI) {
+        String service = System.getenv("CFG_SERVICE_NAME");
+        ServiceType serviceType = new ServiceType(service);
+
+        int retry_limit = 10;
+        for (int i = 1; i <= retry_limit; i++) {
+            System.out.println("Attempting to call service (" + i + "/" + retry_limit + ")");
```

Comment:
Maybe we should say "to call service discovery" to avoid confusing with actually calling the service

Reply:
Otherwise I think it is OK.

Reply:
> Maybe we should say "to call service discovery" to avoid confusing with actually calling the service

Calling it is also done within the same loop. But yeah, this could be improved. I have some more incoming changes in this class that I will put up another PR for, I'll handle it in that one instead.